lib/experimental/class-custom-wp-posts-list-table.php

```diff
@@ -0,0 +1,150 @@
+
+			if ( $lock_holder ) {
+				$lock_holder   = get_userdata( $lock_holder );
+				$locked_avatar = get_avatar( $lock_holder->ID, 18 );
+				/* translators: %s: User's display name. */
+				$locked_text = esc_html( sprintf( __( '%s is currently editing' ), $lock_holder->display_name ) );
+			} else {
+				$locked_avatar = '';
+				$locked_text   = '';
+			}
+
+			echo '<div class="locked-info"><span class="locked-avatar">' . $locked_avatar . '</span> <span class="locked-text">' . $locked_text . "</span></div>\n";
+		}
+
+		$pad = str_repeat( '&#8212; ', $this->current_level );
+		echo '<strong>';
+
+		$title = _draft_or_post_title();
+
+		if ( $can_edit_post && 'trash' !== $post->post_status ) {
+			printf(
+				'<a class="row-title" href="%s" aria-label="%s">%s%s</a>',
+				get_edit_post_link( $post->ID ),
+				/* translators: %s: Post title. */
+				esc_attr( sprintf( __( '&#8220;%s&#8221; (Edit)' ), $title ) ),
+				$pad,
+				$title
+			);
+		} else {
+			printf(
+				'<span>%s%s</span>',
+				$pad,
+				$title
+			);
+		}
+		_post_states( $post );
+
+		// This line is the only difference from the core code.
+		$this->comment_indicator_avatars( $post );
+
+		if ( isset( $parent_name ) ) {
+			$post_type_object = get_post_type_object( $post->post_type );
+			echo ' | ' . $post_type_object->labels->parent_item_colon . ' ' . esc_html( $parent_name );
+		}
+
+		echo "</strong>\n";
+
+		if ( 'excerpt' === $mode
+			&& ! is_post_type_hierarchical( $this->screen->post_type )
+			&& current_user_can( 'read_post', $post->ID )
+		) {
+			if ( post_password_required( $post ) ) {
+				echo '<span class="protected-post-excerpt">' . esc_html( get_the_excerpt() ) . '</span>';
+			} else {
+				echo esc_html( get_the_excerpt() );
+			}
+		}
+
+		/** This filter is documented in wp-admin/includes/class-wp-posts-list-table.php */
+		$quick_edit_enabled = apply_filters( 'quick_edit_enabled_for_post_type', true, $post->post_type );
+
+		if ( $quick_edit_enabled ) {
+			get_inline_data( $post );
+		}
+	}
+
+	/**
+	 * Function to output the avatar HTML for a post ID.
+	 *
+	 * @param int $post The post object.
+	 * @return void
+	 */
+	private function comment_indicator_avatars( $post ) {
+		$unresolved_comments = get_comments(
+			array(
+				'post_id'  => $post->ID,
+				'type'     => 'block_comment',
+				'status'   => 'hold',
+				'per_page' => 100,
+			)
+		);
+
+		// Show indicator for any post with unresolved comments.
+		if ( count( $unresolved_comments ) > 0 ) {
+			$maxAvatars = 3;
+			$count = $maxAvatars;
+			echo "<div class='comment-avatar-stack' title='" . esc_attr__( 'This post has open discussions', 'gutenberg' ) . "'>";
+			foreach ( $unresolved_comments as $comment ) {
+				if ( $count-- <= 0 ) {
+					break;
+				}
+				$gravatar_params = array(
+					'size' => 18,
+					'',
+				);
+				$avatar_urls = get_avatar_url( $comment->user_id, $gravatar_params );
+				echo "<img class='comment-avatar' src='" . esc_url( $avatar_urls ) . "' />";
+			}
```

Comment:
The logic here seems a bit redundant. I think if we use `array_slice`, the decrement logic won't be necessary. For example:

```suggestion
			foreach ( array_slice( $unresolved_comments, 0, $max_avatars ) as $comment ) {
				$gravatar_params = array(
					'size' => 18,
					'',
				);
				$avatar_urls = get_avatar_url( $comment->user_id, $gravatar_params );
				echo "<img class='comment-avatar' src='" . esc_url( $avatar_urls ) . "' />";
			}
```

Reply:
Nice!